src/crafting_gui.cpp

```diff
@@ -150,7 +152,128 @@ static int print_items( const recipe &r, const catacurses::window &w, int ypos,
+        const std::string no_bills = _( "There are no bills here" );
+        if( !bills_vec.empty() ) {
+            draw_scrollbar( w_bills, selection, entries_per_page, bills_vec.size(),
+                            point( 0, 3 ) );
+            for( size_t i = top_of_page; i < bills_vec.size(); i++ ) {
+                const int y = i - top_of_page + 3;
+                trim_and_print( w_bills, point( 1, y ), 43, selection == i ? hilite( col ) : col,
+                                bills_vec[i].obj().result_name() );
+            }
+            if( selection < bills_vec.size() ) {
+                mvwprintz( w_bills, point( 46, 3 ), c_light_gray, _( "Quantity:           %d" ),
+                           all_bills[cur_recipe] );
+            } else {
+                mvwprintz( w_bills, point( 46, 4 ), c_light_red, no_bills );
+            }
+        } else {
+            mvwprintz( w_bills, point( 46, 4 ), c_light_red, no_bills );
+        }
+        mvwprintz( w_bills, point( 1, FULL_SCREEN_HEIGHT - 1 ), c_light_gray,
+                   _( "Press %s to edit bill, %s to create new bill, %s to remove a bill." ),
+                   ctxt.get_desc( "CONFIRM" ), ctxt.get_desc( "NEW_BILL" ), ctxt.get_desc( "REMOVE_BILL" ) );
+        wrefresh( w_bills );
+        const std::string action = ctxt.handle_input();
+        if( action == "DOWN" ) {
+            selection++;
+            if( selection >= bills_vec.size() ) {
+                selection = 0;
+            }
+        } else if( action == "UP" ) {
+            if( selection == 0 ) {
+                selection = bills_vec.empty() ? 0 : bills_vec.size() - 1;
+            } else {
+                selection--;
+            }
+        } else if( action == "CONFIRM" ) {
+            if( !bills_vec.empty() ) {
+                int max_amount = 500;
+                const std::string popupmsg = string_format( _( "Change how many %s to craft ( max 500 )." ),
+                                             bills_vec[selection].obj().result_name() );
+                int amount = string_input_popup()
+                             .title( popupmsg )
+                             .width( 20 )
+                             .text( to_string( bill_here->bills[bills_vec[selection]] ) )
+                             .only_digits( true )
+                             .query_int();
+                if( amount > 0 && amount <= max_amount ) {
+                    bill_here->bills[cur_recipe] = amount;
+                }
+            }
+        } else if( action == "NEW_BILL" ) {
+            int batch_size = 0;
+            const recipe *new_rec = select_crafting_recipe( batch_size, true );
+            g->refresh_all();
+            if( new_rec ) {
+                if( bills_vec.empty() ) {
+                    //create new crafting_bill
+                    crafting_bill new_bill;
+                    std::map<recipe_id, int> new_bills;
+                    new_bills[ new_rec->ident() ] = batch_size;
```

Comment:
```suggestion
                    new_bill.bills[ new_rec->ident() ] = batch_size;
```
Again: no need to make a separate variable that you use only once anyway.